test/run_unittest.sh

```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
 ${CXX} -lm -Wall -O -g ./main_unittest.cxx ../src/bmi_soil_moisture_profile.cxx ../src/soil_moisture_profile.cxx -o run_smp
```

Comment:
required a `-std=c++11` compiler switch (`Apple clang version 14.0.0`)

Reply:
okay, I never tested it on clang so if clang needs we can add it. I mostly use gcc